client/verta/verta/_tracking/experimentrun.py

```diff
@@ -226,32 +226,62 @@ def _log_artifact(self, key, artifact, artifact_type, extension=None, method=Non
                                        "{}://{}/api/v1/modeldb/experiment-run/logArtifact".format(self._conn.scheme, self._conn.socket),
                                        self._conn, json=data)
         if not response.ok:
+            conflict_error = ValueError("artifact with key {} already exists;"
+                                        " consider setting overwrite=True".format(key))
             if response.status_code == 409:
-                raise ValueError("artifact with key {} already exists;"
-                                 " consider setting overwrite=True".format(key))
+                # TODO: check that `artifact_hash` hasn't changed
+
+                # check if multipart upload was in progress
+                url = "{}://{}/api/v1/modeldb/experiment-run/getCommittedArtifactParts".format(
+                    self._conn.scheme,
+                    self._conn.socket,
+                )
+                msg = _CommonService.GetCommittedArtifactParts(id=self.id, key=key)
+                data = _utils.proto_to_json(msg)
+                response = _utils.make_request("GET", url, self._conn, params=data)
+                if response.status_code == 404:  # old backend without multipart support
+                    # TODO: check if this also happens when artifact was not multipart
+                    raise conflict_error
+                elif not response.ok:
+                    _utils.raise_for_http_error(response)
+
+                # get last part number
+                response_msg = _utils.json_to_proto(response.json(), _CommonService.GetCommittedArtifactParts.Response)
+                if not response_msg.artifact_parts:
+                    url_for_artifact = self._get_url_for_artifact(key, "PUT", part_num=1)
+
+                    if url_for_artifact.multipart_upload_ok:  # there should be parts when completed
```

Comment:
Minor: This comment makes more sense to me for understanding this. Feel free to adjust/ignore!
```suggestion
                    if url_for_artifact.multipart_upload_ok:  # multipart artifact logged to MDB, but no parts uploaded
```